pkg/client/unversioned/clientcmd/merged_client_builder_test.go

```diff
@@ -192,3 +194,112 @@ func TestInClusterConfig(t *testing.T) {
 		}
 	}
 }
+
+func TestInClusterConfigNamespace(t *testing.T) {
+	err1 := fmt.Errorf("unique error")
+
+	testCases := map[string]struct {
+		clientConfig *testClientConfig
+		icc          *testICC
+
+		checkedICC bool
+		result     string
+		ok         bool
+		err        error
+	}{
+		"in-cluster checked on other error": {
+			clientConfig: &testClientConfig{err: ErrEmptyConfig},
+			icc:          &testICC{},
+
+			checkedICC: true,
+			err:        ErrEmptyConfig,
+		},
+
+		"in-cluster not checked on non-empty error": {
+			clientConfig: &testClientConfig{err: ErrEmptyCluster},
+			icc:          &testICC{},
+
+			err: ErrEmptyCluster,
+		},
+
+		"in-cluster checked when config is default": {
+			clientConfig: &testClientConfig{},
+			icc:          &testICC{},
+
+			checkedICC: true,
+		},
+
+		"in-cluster not checked when config is not equal to default": {
+			clientConfig: &testClientConfig{namespace: "test", namespaceSpecified: true},
+			icc:          &testICC{},
+
+			result: "test",
+			ok:     true,
+		},
+
+		"in-cluster checked when namespcae is not specified, but is defaulted": {
+			clientConfig: &testClientConfig{namespace: "test", namespaceSpecified: false},
+			icc:          &testICC{},
+
+			checkedICC: true,
+			result:     "test",
+			ok:         false,
+		},
+
+		"in-cluster error returned when config is empty": {
+			clientConfig: &testClientConfig{err: ErrEmptyConfig},
+			icc: &testICC{
+				possible: true,
+				testClientConfig: testClientConfig{
+					err: err1,
+				},
+			},
+
+			checkedICC: true,
+			err:        err1,
+		},
+
+		"in-cluster config returned when config is empty": {
+			clientConfig: &testClientConfig{err: ErrEmptyConfig},
+			icc: &testICC{
+				possible: true,
+				testClientConfig: testClientConfig{
+					namespace:          "test",
+					namespaceSpecified: true,
+				},
+			},
+
+			checkedICC: true,
+			result:     "test",
+			ok:         true,
+		},
+
+		"in-cluster config returned when config is empty with default": {
```

Comment:
With clientcmd it's possible to set default inputs that are applied in the
event the file on disk is empty - basically values coming from environment,
flags, or other sources that are merged with the kubeconfig file.  There's
also the "default namespace" value - end user doesn't specify "default" but
we give it to them.  Will update the key description.

On Fri, Sep 9, 2016 at 5:20 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/client/unversioned/clientcmd/merged_client_builder_test.go
> <https://github.com/kubernetes/kubernetes/pull/32393#discussion_r78251064>
> :
>
> > +		"in-cluster config returned when config is empty": {
> > +			clientConfig: &testClientConfig{err: ErrEmptyConfig},
> > +			icc: &testICC{
> > +				possible: true,
> > +				testClientConfig: testClientConfig{
> > +					namespace:          "test",
> > +					namespaceSpecified: true,
> > +				},
> > +			},
> > +
> > +			checkedICC: true,
> > +			result:     "test",
> > +			ok:         true,
> > +		},
> > +
> > +		"in-cluster config returned when config is empty with default": {
>
> What do you mean by "empty with default"? what is "default" here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/32393/files/76c3add93fded3d159860e3e70b302aec7172c68#r78251064>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p6zIn0zTWFEtqz5qR58vQ9UaPLhLks5qoc2YgaJpZM4J5N0j>
> .
>